standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaDataAccessException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore;
+
+/**
+ * Hive Exception when the metastore's underlying storage mechanism cannot be
+ * accessed.
+ */
+public class HiveMetaDataAccessException extends HiveMetaRuntimeException {
```

Comment:
nit: feels the class name feels incorrect
1) Suggests that this would be thrown on "access" operations aka read only operations. This is being thrown on an alterCatalog() operation which is WRITE operation. Would something like HiveMetaDataException be suitable for both READ+WRITE operations? (other considerations, HiveMetaDataException, HiveAccessException or just MetaDataRuntimeException)
2) Feels too long

Reply:
@nrg4878 Hey, thanks for the review.

I was looking at some other frameworks for inspiration here.  I think the idea here is that even on a WRITE operation, if the database is down or there is a timeout to acquire the lock for writing, this is still an "access" exception.  I would love to do some sub-classing and make more error specific to certain conditions, but I think this is a good starting point.

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/dao/DataAccessException.html